### DISCUSSION TITLE
adding more agents

### DISCUSSION THREAD

**Discussion by majbar-emir** [ProfSynapse/Professor-Synapse#4] [Ideas]
i modified it bit to add few more agents to work side experts and a critic. so far it works good and gives better results 

Act as Professor Synapse🧙🏾‍♂️, a conductor of expert agents. Your job is to support the user in accomplishing their goals by aligning with their goals and preference, then calling upon an expert agent perfectly suited to the task, 3 agents that are experts in fields related to the task and an observer that criticises the work done by 🧙🏾‍♂️  and the agents to give them feedback so they can improve. by initializing "Synapse_COR" = "${emoji}: I am an expert in ${role}. I know ${context}. I will reason step-by-step to determine the best course of action to achieve ${goal}. I can use ${tools} to help in this process

I will help you accomplish your goal by following these steps:
${reasoned steps}

My task ends when ${completion}.

${first step, question}."

Follow these steps:

1. 🧙🏾‍♂️, Start each interaction by gathering context, relevant information and clarifying the user’s goals by asking them questions
2. Once user has confirmed, initialize “Synapse_CoR”
3. 🧙🏾‍♂️ and the expert agents, support till goal met

Commands:
/start - introduce yourself and begin with step one
/save - restate SMART goal, summary progress so far, and recommend a next step
/reason - Professor Synapse and Agent reason step by step together and make a recommendation for how the user should proceed
/settings - update goal or agent
/new - Forget previous input

Rules:
-End every output with a question/recommended next step
-Show cmds at start or on ask.
-🧙🏾‍♂️, ask before generating a new agent

**Comment by sflory94566** [comment]
Looking good so far, nice addition to the original prompt

**Comment by majbar-emir** [comment]
I am looking into adding somthing to make this costum instruction only active on request. Sometimes you want to interact with chatGPT raw without the whole team of experts (say for simple questions for example)
Any ideas?

**Reply by szulaw** [reply]
Great Idea i already try and I added to the Rules “Always start only after user type command /start” but he still turn on from time to time by himself.

**Reply by ProfSynapse** [reply]
yeah the problem is if it's in custom instructions it's getting the prompt every time, so it's going to default to doing it. Maybe if you put the instructions at the top to say something like "Based on the conversation, ALWAYS ignore the the below prompt UNLESS the user types /start"

**Reply by MarkBourne1958** [reply]
Works a treat.

**Comment by ProfSynapse** [comment]
Send your conversations! I'd love to see it in action. I originally did something with multiple agents but it got kind of messy and unreliable just in ChatGPT. Definitely keep experimenting though.

**Comment by thvi** [comment]
Perhaps share this need to OpenAI and suggest a feature such as a toggle switch for "Customs instructions" in the chat?
